src/controller/controller.class.ts

```diff
@@ -0,0 +1,73 @@
+import { NextFunction, Request, Response } from 'express';
+import { httpStatusCode, errorWrapper } from '../utils';
+import { Service } from '../service/service.class';
+
+export class Controller<TEntity> {
+  getAllItems: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  getItemById: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  getItemByWhere: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  createItem: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  createItems: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  updateItem: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  deleteItem: (_req: Request, _res: Response, _next: NextFunction) => Promise<void>;
+
+  constructor(service: Service<TEntity>) {
+    this.getAllItems = errorWrapper(async (req: Request, res: Response) => {
+      const allItems = await service.getAllItems();
+
+      if (Array.isArray(allItems.data) && allItems.data.length === 0)
+        return res.status(httpStatusCode.NO_CONTENT).json(allItems);
+
+      res.status(httpStatusCode.OK).json(allItems);
+    });
+
+    this.getItemById = errorWrapper(async (req: Request, res: Response) => {
+      const item = await service.getItemById(req.params.id);
+
+      if (!item) return res.status(httpStatusCode.BAD_REQUEST).json({ message: 'Item not found' });
+
+      res.status(httpStatusCode.OK).json(item);
+    });
+
+    this.getItemByWhere = errorWrapper(async (req: Request, res: Response) => {
+      const item = await service.getItemByWhere(req.body);
+
+      if (!item) return res.status(httpStatusCode.BAD_REQUEST).json({ message: 'Item not found' });
+
+      res.status(httpStatusCode.OK).json(item);
+    });
+
+    this.createItem = errorWrapper(async (req: Request, res: Response) => {
+      const createdItem = await service.createItem(req.body);
+
+      res.status(httpStatusCode.CREATED).json(createdItem);
+    });
+
+    this.createItems = errorWrapper(async (req: Request, res: Response) => {
+      const createdItems = await service.createItems(req.body);
+
+      res.status(httpStatusCode.CREATED).json(createdItems);
+    });
+
+    this.updateItem = errorWrapper(async (req: Request, res: Response) => {
```

Comment:
Can we rename these to updateItemById and deleteItemById ? Maybe even add a update and delete item by where

Reply:
I haven't finalized the default functionality yet, but I also want users to be able to add their own endpoint by configuration as well.

changed the names of update and delete by Id